block-producer/src/block_builder/errors.rs

```diff
@@ -4,18 +4,25 @@ use thiserror::Error;
 
 use crate::store::{ApplyBlockError, BlockInputsError};
 
+use super::prover::CREATED_NOTES_TREE_INSERTION_DEPTH;
+
 #[derive(Debug, Error, PartialEq, Eq)]
 pub enum BuildBlockError {
-    #[error("failed to update account root: {0}")]
-    AccountRootUpdateFailed(#[from] BlockProverError),
     #[error("failed to apply block: {0}")]
     ApplyBlockFailed(#[from] ApplyBlockError),
+    #[error("failed to compute new block header: {0}")]
+    ComputeBlockHeaderFailed(#[from] BlockProverError),
     #[error("failed to get block inputs from store: {0}")]
     GetBlockInputsFailed(#[from] BlockInputsError),
     #[error("transaction batches and store don't modify the same account IDs. Offending accounts: {0:?}")]
     InconsistentAccountIds(Vec<AccountId>),
     #[error("transaction batches and store contain different hashes for some accounts. Offending accounts: {0:?}")]
     InconsistentAccountStates(Vec<AccountId>),
+    #[error(
+        "Too many batches in block. Got: {0}, max: 2^{}",
+        CREATED_NOTES_TREE_INSERTION_DEPTH
+    )]
+    TooManyBatchesInBlock(usize),
```

Comment:
minor nit: seems like this is the only error variant where the message is capitalized. We should probably keep things consistent.

Reply:
done